src/main/java/org/myrobotlab/service/LcdDisplay.java

```diff
@@ -0,0 +1,232 @@
+  private final byte LCD_BACKLIGHT = (byte) 0x08;
+  private final byte LCD_NOBACKLIGHT = (byte) 0x00;
+
+  private final byte En = (byte) 0b00000100; // Enable bit
+  private final byte Rw = (byte) 0b00000010; // Read/Write bit
+  private final byte Rs = (byte) 0b00000001; // Register select bit
+
+  private boolean backLight;
+  public Map<Integer, String> screenContent = new HashMap<Integer, String>();
+
+  public void attach(Pcf8574 pcf8574) {
+    // we need more checkup here...
+    if (pcf8574.isAttached) {
+      this.i2cControler = pcf8574;
+      setReady(true);
+      log.info("{} attach {}", getName(), pcf8574.getName());
+    } else {
+      log.error("Cannot attach {} to {}", getName(), pcf8574.getName());
+    }
+  }
+
+  /*
+   * INIT LCD panel
+   */
+  public void init() {
+    log.info("Init I2C Display");
+    lcd_write((byte) 0x03);
+    lcd_write((byte) 0x03);
+    lcd_write((byte) 0x03);
+    lcd_write((byte) 0x02);
+    lcd_write((byte) (LCD_FUNCTIONSET | LCD_2LINE | LCD_5x8DOTS | LCD_4BITMODE));
+    lcd_write((byte) (LCD_DISPLAYCONTROL | LCD_DISPLAYON));
+    lcd_write((byte) (LCD_CLEARDISPLAY));
+    lcd_write((byte) (LCD_ENTRYMODESET | LCD_ENTRYLEFT));
+  }
+
+  /*
+   * Send byte to PCF controller
+   */
+  public void write_cmd(byte cmd) {
+    if (isReady()) {
+      i2cControler.writeRegister(cmd);
+    } else {
+      log.error("LCD is not ready / attached !");
+    }
+  }
+
+  /*
+   * Turn ON/OFF LCD backlight
+   */
+  public void setBackLight(boolean status) {
+    if (status == true) {
+      write_cmd(LCD_BACKLIGHT);
+    } else {
+      write_cmd(LCD_NOBACKLIGHT);
+    }
+    backLight = status;
+    log.info("LCD backlight set to {}", status);
+    broadcastState();
+  }
+
+  public boolean getBackLight() {
+    return backLight;
+  }
+
+  /*
+   * Display string on LCD, by line 
+   */
+  public void display_string(String string, int line) {
+    screenContent.put(line, string);
+    broadcastState();
+    switch (line) {
+      case 1:
+        lcd_write((byte) 0x80);
+        break;
+      case 2:
+        lcd_write((byte) 0xC0);
+        break;
+      case 3:
+        lcd_write((byte) 0x94);
+        break;
+      case 4:
+        lcd_write((byte) 0xD4);
+        break;
+    }
+
+    for (int i = 0; i < string.length(); i++) {
+      lcd_write((byte) string.charAt(i), Rs);
+    }
+  }
+
+  private void lcd_strobe(byte data) {
+    i2cControler.writeRegister((byte) (data | En | LCD_BACKLIGHT));
+    i2cControler.writeRegister((byte) ((data & ~En) | LCD_BACKLIGHT));
+  }
+
+  private void lcd_write_four_bits(byte data) {
+
+    i2cControler.writeRegister((byte) (data | LCD_BACKLIGHT));
+    lcd_strobe(data);
+  }
+
+  private void lcd_write(byte cmd, byte mode) {
+    lcd_write_four_bits((byte) (mode | (cmd & 0xF0)));
+    lcd_write_four_bits((byte) (mode | ((cmd << 4) & 0xF0)));
+  }
+
+  // write a command to lcd
+  private void lcd_write(byte cmd) {
+    lcd_write(cmd, (byte) 0);

```

Comment:
all above should be standard Java camel case method names lcdWrite lcdStrobe lcdWriteFourBits ... etc..